src/bitcoin-cli.cpp

```diff
@@ -307,23 +396,24 @@ int CommandLineRPC(int argc, char *argv[])
             while (std::getline(std::cin,line))
                 args.push_back(line);
         }
+        std::unique_ptr<BaseRequestHandler> rh;
+        if (gArgs.GetBoolArg("-getinfo", false)) {
+            rh.reset(new GetinfoRequestHandler());
+            args.clear();
+            args.push_back("getinfo");
```

Comment:
I think it would be clearer to not reset `args` here. The only reason you're doing this is to pass the `if (args.size() < 1)` check below and then extract `strMethod`.

Instead, just move the `if (args.size() < 1)` check into the else branch. `GetinfoRequestHandler.PrepareRequest()` doesn't require `strMethod` or `args`.

Taking this further, you could add a method to `DefaultRequestHandler` to update the `strMethod` and `args`, and then update the `PrepareRequest()` method to not take any arguments.